Week-1/assignment-1/README.md

```diff
@@ -0,0 +1,83 @@
+
+The request line contains the following:
+
+- a request method, which is one of GET, POST, PUT, PATCH, DELETE, or a handful of other HTTP verbs
+- the path, pointing to the requested resources
+- the HTTP version to communicate with
+
+The next part of the request is the request headers. Headers pass extra information along from the client that help route the request, indicate what type of client is making the request (the user agent), and can be used for supporting A/B testing, blue/green deployments, and canary deployments.
+Headers are key-value pairs like "Accept : */*"
+
+
+## Server processes request and sends back a response
+The server takes the request and based on the info in the request line, headers, and body, decides how to process the request. For the request, GET /about-us/ HTTP/1.1, the server gets the content at this path, constructs the response and sends it back to the client. The response contains the following:
+
+a status line, telling the client the status of the request
+response headers, telling the browser how to handle the response
+the requested resource at that path, either content like HTML, CSS, Javascript, or image files, or data
+The status line contains both the HTTP version and a numeric and text representation of the status of the request.
+
+The browser considers a status code in the 200s to be successful. The response was 200, so the browser knows to render the response.
+Resources can be static files, either text (i.e index.html) or non-text data (i.e. logo.img). Browsers aren’t always requesting static files, though. Often, these are dynamic resources generated at the time of the request and there is no file associated with the resource. In fact, in this request, that’s exactly what is happening. There is no file hello-world, but the server still knows how to process the request. The server generates a dynamic resource, building the HTML from fragments or templates and combining it with dynamic data to send back in the response, as text, so the browser can render the page.
+Now that you know how the server builds the response to send back to the browser, let’s take a look at how the browser handles the response.
+
+
+## 6. Browser renders the content
+Once the browser has received the response from the server, it inspects the response headers for information on how to render the resource. The Content-Type header above tells the browser it received an HTML resource in the response body. The browser knows what to do with HTML!
+The first GET request returns HTML, the structure of the page. But if you inspect the HTML of the page (or any web page) with your browser’s dev tools, you’ll see it references other Javascript, CSS, image resources and requests additional data in order to render the web page as designed.
+As the browser is parsing and rendering the HTML, it is making additional requests to get Javascript, CSS, images, and data
```

Comment:
@chinmay4o no explanation of rendering engine, components of browser, parsers, script execution provided in the README. Please update